python/src/opendp/_convert.py

```diff
@@ -364,3 +367,41 @@ def _slice_to_hashmap(raw: FfiSlicePtr) -> Dict[Any, Any]:
 
 def _wrap_in_slice(ptr, len_: int) -> FfiSlicePtr:
     return FfiSlicePtr(FfiSlice(ctypes.cast(ptr, ctypes.c_void_p), len_))
+
+
+#                            (output         , input       )
+CallbackFn = ctypes.CFUNCTYPE(ctypes.c_void_p, AnyObjectPtr)
+
+def _wrap_py_func(func, TO):
+    from opendp._convert import c_to_py, py_to_c
+
+    def wrapper_func(c_arg):
+        try:
+            # 1. convert AnyObject to Python type
+            py_arg = c_to_py(c_arg)
+            # don't free c_arg, because it is owned by Rust
+            c_arg.__class__ = ctypes.POINTER(AnyObject)
+
+            # 2. invoke the user-supplied function
+            py_out = func(py_arg)
+
+            # 3. convert back to an AnyObject
+            c_out = py_to_c(py_out, c_type=AnyObjectPtr, type_name=TO)
+            # don't free c_out, because we are giving ownership to Rust
+            c_out.__class__ = ctypes.POINTER(AnyObject)
+
+            # 4. pack up into an FfiResult
+            lib.ffiresult_ok.argtypes = [ctypes.c_void_p]
+            lib.ffiresult_ok.restype = ctypes.c_void_p
+            return lib.ffiresult_ok(ctypes.addressof(c_out.contents))
```

Comment:
Aren't you leaking the `FFIResult` here?

Reply:
It shouldn't, because Rust takes ownership of the `FfiResult` here:

https://github.com/opendp/opendp/pull/572/files#diff-d59f16e57c55ad474cc8d65373f333ede766806b239443d5bdfed23f44ff42a4R31-R33

Reply:
Ah yes, I see it now, thanks!